packages/transaction-controller/src/hooks/SequentialPublishBatchHook.ts

```diff
@@ -0,0 +1,227 @@
+import type EthQuery from '@metamask/eth-query';
+import { rpcErrors } from '@metamask/rpc-errors';
+import { createModuleLogger } from '@metamask/utils';
+import type { Hex } from '@metamask/utils';
+
+import type { PendingTransactionTracker } from '../helpers/PendingTransactionTracker';
+import { projectLogger } from '../logger';
+import {
+  type PublishBatchHook,
+  type PublishBatchHookRequest,
+  type PublishBatchHookResult,
+  type TransactionMeta,
+} from '../types';
+
+const log = createModuleLogger(projectLogger, 'sequential-publish-batch-hook');
+
+type SequentialPublishBatchHookOptions = {
+  publishTransaction: (
+    ethQuery: EthQuery,
+    transactionMeta: TransactionMeta,
+  ) => Promise<Hex>;
+  getTransaction: (id: string) => TransactionMeta;
+  getEthQuery: (networkClientId: string) => EthQuery;
+  getPendingTransactionTrackerByChainId: (
```

Comment:
My fault, this should be `getPendingTransactionTrackerByNetworkClientId` or just `getPendingTransactionTracker` as shorter?

Reply:
Make sense, changed to `getPendingTransactionTracker`